symfony/src/Controller/Website/UserController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace App\Controller\Website;
+
+use App\Entity\User;
+use App\Form\PasswordResetType;
+use App\Repository\ProfileRepository;
+use App\Service\Website\UserService;
+use App\Utils\CompanyHelper;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class UserController
+ * @package App\Controller\Website
+ * @Route("/user", name="icapps_website.user")
+ */
+class UserController extends AbstractController
+{
+
+    public function __construct(private TranslatorInterface $translator, private UserService $userService, private ProfileRepository $profileRepository, private CompanyHelper $companyHelper)
+    {
+    }
+
+    /**
+     * User confirmation/activation
+     * @Route("/confirmation/{token}", name=".confirmation")
+     * @param string $token
+     * @return Response
+     */
+    public function confirmation(string $token): Response
+    {
+        $user = $this->userService->activateUser($token);
+
+        return $this->confirmationResponse($user);
+    }
+
+    /**
+     * User pending email confirmation
+     * @Route("/confirmation/pending-email/{token}", name=".confirmation.pending_email")
+     * @param string $token
+     * @return Response
+     */
+    public function confirmationPendingEmail(string $token): Response
+    {
+        $user = $this->userService->activatePendingEmailOfUser($token);
+
+        return $this->confirmationResponse($user);
+    }
+
+    /**
+     * User password reset.
+     * @Route("/password-reset/{token}", name=".reset")
+     * @param string $token
+     * @param Request $request
+     * @return Response
+     */
+    public function passwordReset(string $token, Request $request): Response
+    {
+        // Validate reset token.
+        $user = $this->userService->validatePasswordResetToken($token);
+        if (!$user) {
+            return $this->render('general/status.html.twig', [
+                'companyData' => $this->companyHelper->getBaseCompanyInfo(),
```

Comment:
Best companyData hernoemen dat het duidelijk is dat dit over styling templates gaat. Nu lijkt dit gerelateerd aan user/account.

Is er geen betere manier om die templates in company style te renderen? Niet relevant om dat telkens mee te moeten geven.

Reply:
Zal nu met de get_env() Twig functie werken, veel cleaner